tests/client.go

```diff
@@ -0,0 +1,736 @@
+package tests
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/flux/csv"
+	"github.com/influxdata/influxdb/v2"
+	influxhttp "github.com/influxdata/influxdb/v2/http"
+	"github.com/influxdata/influxdb/v2/pkg/httpc"
+)
+
+type ClientConfig struct {
+	UserID             influxdb.ID
+	OrgID              influxdb.ID
+	BucketID           influxdb.ID
+	DocumentsNamespace string
+
+	// If Session is provided, Token is ignored.
+	Token   string
+	Session *influxdb.Session
+}
+
+// Client provides an API for writing, querying, and interacting with
+// gateway's resources like authorizations, buckets, and organizations.
+type Client struct {
+	Client *httpc.Client
+	*influxhttp.Service
+
+	ClientConfig
+}
+
+// NewClient initialises a new Client which is ready to write points to the Gateway write endpoint.
+func NewClient(gatewayURL string, config ClientConfig) (*Client, error) {
+	opts := make([]httpc.ClientOptFn, 0)
+	if config.Session != nil {
+		config.Token = ""
+		opts = append(opts, httpc.WithSessionCookie(config.Session.Key))
+	}
+	hc, err := influxhttp.NewHTTPClient(gatewayURL, config.Token, false, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, err := influxhttp.NewService(hc, gatewayURL, config.Token)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		Client:       hc,
+		Service:      svc,
+		ClientConfig: config,
+	}, nil
+}
+
+// Open opens the client
+func (c *Client) Open() error { return nil }
+
+// Close closes the client
+func (c *Client) Close() error { return nil }
+
+// MustWriteBatch calls WriteBatch, panicking if an error is encountered.
+func (c *Client) MustWriteBatch(points string) {
+	if err := c.WriteBatch(points); err != nil {
+		panic(err)
+	}
+}
+
+// WriteBatch writes the current batch of points to the Gateway HTTP endpoint.
+func (c *Client) WriteBatch(points string) error {
+	return c.WriteService.Write(context.Background(), c.OrgID, c.BucketID, strings.NewReader(points))
+}
+
+// WriteLegacyBatch writes the current batch of points to the legacy influx 1.x
+// /write endpoint.  Used to mimic our cloud 1 to cloud 2 go replay forwarding.
+func (c *Client) WriteLegacyBatch(org, db, rp, points string) error {
+	writeFn := func(w io.Writer) (string, string, error) {
+		_, err := fmt.Fprint(w, points)
+		return "", "", err
+	}
+	req := c.Client.Post(writeFn, WritePathLegacy).
+		QueryParams([2]string{"db", db}).
+		QueryParams([2]string{"rp", rp}).
+		QueryParams([2]string{"org", org}).
+		StatusFn(httpc.StatusIn(http.StatusNoContent))
+
+	return req.Do(context.Background())
+}
+
+// Query returns the CSV response from a flux query to gateway.
+//
+// This also remove all the \r to make it easier to write tests.
+func (c *Client) QueryFlux(org, query string) (string, error) {
```

Comment:
I don't see this used anywhere. It looks useful for future tests, but not sure it makes sense under the `_v1` suites. WDYT?

Reply:
This is in the tests package, which is shared, so it is possible we will be used for test scenarios using Flux.